drafts/draft-0005.md

```diff
@@ -0,0 +1,142 @@
+data entries, and provides guideline about how to parse them and to handle
+edge cases.
+
+## Motivation
+
+As a generic mechanism to store key/value pairs on the ledger, data entries 
+accept a very wide range of use cases. If each actor use it following its own 
+syntax, it will lead to a confusing experience and ultimatelly to naming 
+conflicts.
+
+Introducing namespaces allow to hierarchize and organize data, in such a way 
+that sorting the key/value entries will result in a readable output even when 
+using them for various purpose on the same account. It also helps to prevent
+name clash.
+
+The present SEP ensure that all other SEP that will use data entries will do so 
+in a consistent way. It provides guidelines ontop of which generic tool to 
+handle data entries can be created.
+
+## Specification
+
+Namespace hierarchy is encoded in the key using one or more terms separated by 
+dots. Terms are made of lowcase letters, undescore and numbers and musn't start
+with a number.
+
+**Examples**
+
+```
+conf.two_factor
+conf.multisig.server
+conf.multisig.network
+conf.multisig.key
+profile.name
+profile.language
+wallet.btc
+wallet.eth
+```
+
+* **Term REGEXP:** `[_a-z][_a-z0-9]*`
+* **Key REGEXP:** `[_a-z][_a-z0-9]*(\.[_a-z][_a-z0-9]*)*`
+
+When parsing the data tree in a given programming language, keys that doesn't 
+respect this syntax must be ignored and valid entries must be hierarchized in 
+an adequate data structure. If a term is used both as an identifier and a 
+namespace, a warning must be thrown by the parser and the identifier value must 
+be ignored.
+
+
+**Example: same data tree in JSON**
+
+```js
+{
+  conf: {
+    two_factor: ...,
+    multisig: {
+      server: ...,
+      network: ...,
+      key: ...
+    }
+  },
+  profile: {
+    name: ...,
+    language: ...
+  },
+  wallet: {
+    btc: ...,
+    eth: ...
+  }
+}
+```
+
+
+## Rationale
+
+**Syntax**
+
+The namespace syntax had to be familiar, and terms have to be written in a way
+that is valid accross a broad range of programming languages.
+
+This protocol uses JavaScript syntax because it is the language the most 
+developers uses. It is also familiar to anybody using JSON, which is a format 
+widely used to pass data, and the one already used by horizon API.
+
+Parsing the data tree must lead to the same namespace/key/value structure 
+regardless of the programming language being used. This is the reason why term 
+syntax must be restricted, as any unusual character could lead to errors in 
+this or that language. The lowcase restiction is meant to prevent a different 
+data tree interpretation between case-sensitive and case-insensitive languages.
+
+Forbidding the use of numbers and/or undescore is advisable if it appears that 
+they prevent the data tree to be properly parsed in any well known programming 
+language.
+
+**Identifier/namespace conflict**
+
+This conflict arise when a term is used both as an identifier and a namespace, 
```

Comment:
This sort of conflict is why I feel there should be less emphasis on turning it into a data structure and more just some loose guidelines about what are the namespace delimiter and what things are considered "under" the namespace. That way you can just call out some "known namespaces" and not really worry about the details like this.

Reply:
It's true I could have chosen that strategy. Turning into data structure is not a must have: it is a very convenient feature for devs that comes at a no cost with this design. It asked me an additional effort to put it right but that's it.

Now I assume that by saying *this sort of confict* you're introducing the idea that there are other possible conflict that are not properly handled by this SEP, and that turning into data structure might require additional effort. Can you elaborate on that?

Because, if there's no other kind of conflict and everything works fine, I don't see any valid reason to drop this feature.

Reply:
No, I just don't think it should be part of the SEP at all since it is application/use case specific how you might want to use these namespaces/whatever data structure you decided to build out of them.

Reply:
The usage of a namespace might or might not be application specific. The way to write them obviously require a standard.